src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -67,12 +74,19 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             if (AppendVersion)
             {
                 EnsureFileVersionProvider();
+
+                string resolvedUrl;
+                if (TryResolveUrl(Src, tryEncodeApplicationPath: false, resolvedUrl: out resolvedUrl))
+                {
+                    Src = resolvedUrl;
+                }
                 output.Attributes[SrcAttributeName] = _fileVersionProvider.AddFileVersionToPath(Src);
```

Comment:
Existing issue but this doesn't play nicely with earlier tag helpers. Should look similar to other changed tag helpers e.g.
``` c#
    if (Src != null)
    {
        output.CopyHtmlAttribute(SrcAttributeName, context);

        // Resolve any application relative URLs (~/) now so they can be used in comparisons later.
        if (TryResolveUrl(Src, encodeWebRoot: false, resolvedUrl: out resolvedUrl))
        {
            Src = resolvedUrl;
        }
    }

    ProcessUrlAttribute(SrcAttributeName, output);
```

Note I moved `ProcessUrlAttribute` call out of the conditional block. This allows another tag helper to add a `src` attribute and have this one fix it up.